### PR TITLE
refactor(notifications): extract helpers from sendTestEmail

### LINKED ISSUE #44: The `sendTestEmail` method exceeds 20 lines, violating the single-purpose method length guideline. Extract setup, validation, and response formatting into helper functions for clarity and maintainability.
The `sendTestEmail` method exceeds 20 lines, violating the single-purpose method length guideline. Extract setup, validation, and response formatting into helper functions for clarity and maintainability.

_Originally posted by @Copilot in https://github.com/ChrisRomp/playa-plan/pull/42#discussion_r2117238309_

### PR DESCRIPTION
**Pull request by ChrisRomp**
## Motivation

The `sendTestEmail` method in both the service (~42 lines) and controller (~97 lines) exceeded the project's ~20-line guideline and violated the "thin controllers" principle. The controller also had duplicated response-building logic and parsed recipients twice.

## Approach

Decomposed both methods into focused, single-purpose helpers:

**Service** -- main method reduced to ~15 lines:
- `parseRecipients()` -- splits comma-separated input, guards against empty list
- `validateEmailAddresses()` -- regex validation per address
- `sendTestEmailToRecipient()` -- single-recipient send with error handling
- Return type changed from `boolean` to `TestEmailResult { success, recipients }` so the controller doesn't need to independently parse recipients

**Controller** -- main method reduced to ~20 lines:
- `validateEmailConfiguration()` -- checks email enabled + SMTP configured
- `buildTestEmailDetails()` -- assembles config from multiple sources
- `buildTestEmailResponse()` -- single response builder for both success/failure (eliminates duplicated 18-line response objects)

**Interfaces** added in `notifications/interfaces/test-email.interfaces.ts`:
- `TestEmailDetails`, `TestEmailCustomContent`, `TestEmailResult`, `TestEmailResponse`

## Bug fixes included

- Empty recipients (e.g. `""`) previously returned `true` silently with 0 sends. Now throws with a clear message.
- Used `??` instead of `||` for `smtpPort` and `smtpUseSsl` defaults (port 0 is valid; `false` is meaningful for SSL).

## Verification

- All 32 notification module tests pass
- Zero TypeScript errors (`tsc --noEmit`)
- Zero lint errors

## Follow-up issues filed

- #168 -- Sanitize HTML in email templates
- #169 -- SMTP validation should check username/password
- #170 -- Add unit tests for new edge cases

Closes #44

### CODE REVIEW

File: apps/api/src/notifications/controllers/notifications.controller.ts [hunk 1]
```diff
@@ -232,6 +148,75 @@ export class NotificationsController {
...
+  /**
+   * Validate that email sending is enabled and SMTP is configured.
+   * @throws Error if configuration is invalid
+   */
+  private validateEmailConfiguration(emailConfig: { emailEnabled?: boolean; smtpHost?: string | null; senderEmail?: string | null }): void {
+    if (!emailConfig.emailEnabled) {
+      throw new Error('Email notifications are currently disabled. Please enable email notifications first.');
+    }
+    if (!emailConfig.smtpHost || !emailConfig.senderEmail) {
+      throw new Error('SMTP configuration is incomplete. Please configure SMTP settings before sending test emails.');
```
[the remark this answers is not in the document]
Comment (ChrisRomp) [reply apps/api/src/notifications/controllers/notifications.controller.ts]: Accepted. Added `smtpUsername` and `smtpPassword` checks to `validateEmailConfiguration()` so the endpoint fails fast with a clear "SMTP authentication credentials are not configured" message instead of proceeding to a generic send failure.

File: apps/api/src/notifications/services/notifications.service.ts [hunk 2]
```diff
@@ -352,69 +353,72 @@ export class NotificationsService {
...
-      timestamp: Date;
-    },
+    testEmailDetails: TestEmailDetails,
     userId: string,
-    customContent?: {
-      subject?: string;
-      message?: string;
-      format?: 'html' | 'text';
-      includeSmtpDetails?: boolean;
+    customContent?: TestEmailCustomContent
```
[the remark this answers is not in the document]
Comment (ChrisRomp) [reply apps/api/src/notifications/services/notifications.service.ts:362]: Accepted. Added 6 unit tests covering: empty string, whitespace-only input, invalid email, single valid recipient, comma-separated multi-recipient, and partial failure scenarios. All pass.
